### PR TITLE
[jit_kernel] Add JIT ngram_utils kernel   

### PR DESCRIPTION
**Pull request by Johnsonms**
## Motivation
https://github.com/sgl-project/sglang/issues/17865
The reconstruct_indices_from_tree_mask kernel in sgl-kernel reconstructs
tree traversal indices (positions, retrive_index, retrive_next_token,
retrive_next_sibling) from a flat boolean tree attention mask. Porting it
to the JIT kernel framework makes it available without requiring a
pre-compiled sgl-kernel package, improving portability and development
iteration speed.

## Modifications
 - python/sglang/jit_kernel/csrc/speculative/ngram_utils.cuh: JIT port of
  sgl-kernel/csrc/speculative/ngram_utils.cu. Replaces PyTorch ATen tensor
  types with tvm::ffi::TensorView, at::cuda::getCurrentCUDAStream() with
  LaunchKernel::resolve_device(), and adds comprehensive input validation via
   RuntimeCheck. The CUDA device kernel reconstructIndicesFromTreeMask is
  unchanged.
  - python/sglang/jit_kernel/ngram_utils.py: Python wrapper for
  reconstruct_indices_from_tree_mask using @register_custom_op, following the
   same pattern as other JIT kernels.
  - python/sglang/jit_kernel/tests/test_ngram_utils.py: Tests covering smoke
  runs across batch sizes and tree sizes, known-answer tests for a
  linear-chain tree (positions, next_token links, sibling links), and JIT vs
  AOT cross-validation (15 tests, all passing).
  - python/sglang/jit_kernel/benchmark/bench_ngram_utils.py:
  triton.testing.perf_report benchmark comparing JIT vs AOT across typical
  batch sizes and draft token counts, with a correctness diff section.

## Accuracy Tests
`python -m pytest python/sglang/jit_kernel/tests/test_ngram_utils.py `

image

## Benchmarking and Profiling
`python python/sglang/jit_kernel/benchmark/bench_ngram_utils.py`
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Comment by hnyls2002** [comment]
Thanks @Johnsonms! The target tree `python/sglang/jit_kernel/` no longer exists - it was retired into `python/sglang/kernels/` (#32072, #33400) - and the tracking issue #17865 is closed. `reconstruct_indices_from_tree_mask` is still AOT-only, imported from `sgl_kernel.speculative` at `python/sglang/srt/speculative/ngram_worker.py:6`, so a fresh port against `python/sglang/kernels/jit/` would still be welcome. Closing as obsolete - please reopen if I've missed something.
